pkg/controller/replication_controller.go

```diff
@@ -257,14 +257,18 @@ func (rm *ReplicationManager) deletePod(obj interface{}) {
 		}
 	}
 	if rc := rm.getPodControllers(pod); rc != nil {
-		rm.expectations.DeletionObserved(rc)
+		rcKey, err := controllerKeyFunc(rc)
+		if err != nil {
+			glog.Errorf("Couldn't get key for replication controller %+v: %v", rc, err)
```

Comment:
Perhaps consider pushing the key function down inside the expectations object, so that you don't have to check this error in so many locations?